cpp/dolfinx/mesh/Topology.cpp

```diff
@@ -442,6 +442,59 @@ std::vector<bool> mesh::compute_boundary_facets(const Topology& topology)
   return _boundary_facet;
 }
 //-----------------------------------------------------------------------------
+std::vector<bool> mesh::compute_interface_facets(const Topology& topology)
+{
+  const int tdim = topology.dim();
+  auto facets = topology.index_map(tdim - 1);
+  if (!facets)
+    throw std::runtime_error("Facets have not been computed.");
+
+  auto fc = topology.connectivity(tdim - 1, tdim);
+  if (!fc)
+    throw std::runtime_error("Facet-cell connectivity missing.");
+
+  const std::vector<int32_t>& fwd_shared_facets
+      = facets->scatter_fwd_indices().array();
+
+  std::int32_t num_facets = facets->size_local() + facets->num_ghosts();
+  std::vector<bool> _interface_facets(num_facets, false);
+
+  // If a shared facet is connected to a single cell it's a candidate to be on
+  // the interface between adjacent subdomains
+  for (const std::int32_t& f : fwd_shared_facets)
+    _interface_facets[f] = fc->num_links(f) == 1;
+
+  // If a ghost facet is connected to a single cell it's a candidate to be on
+  // the interface between adjacent subdomains
+  for (std::int32_t f = facets->size_local(); f < num_facets; ++f)
+    _interface_facets[f] = fc->num_links(f) == 1;
+
+  // Remove local exterior domain facets from candidates
+  // Fixme: This is problably unnecessary and going to be removed.
+  std::vector<bool> boundary_facet = mesh::compute_boundary_facets(topology);
+  std::transform(boundary_facet.begin(), boundary_facet.end(),
+                 _interface_facets.begin(), _interface_facets.begin(),
+                 [](const auto& f0, const auto& f1)
+                 { return f0 ? false : f1; });
+
+  // Remove remote exterior domain facets from candidates
+  // Note: std::vector<bool> cannot be converted to span<bool>, using
```

Comment:
Feel free to change functions that return `std::vector<bool>` to `std::vector<std::int8_t>` if it makes code simpler. It will probably be faster too.